doc/tutorials/basic/pn_bind.md

```diff
@@ -1,9 +1,332 @@
+)
+
+submit = pn.widgets.Button(name="Submit")
+
+def result(clicked):
+    if clicked:
+        return calculate_power_fn()
+    return "Click Submit"
+
+result_fn = pn.bind(result, submit)
+
+pn.Column(
+    wind_speed, efficiency, submit, result_fn
+).servable()
+```
+
+## Harnessing Throttling for Performance
+
+To prevent excessive updates and ensure smoother performance, you can apply throttling. This limits the rate at which certain actions or events occur, maintaining a balanced user experience:
+
+```{pyodide}
+import panel as pn
+from time import sleep
+
+pn.extension()
+
+
+def calculate_power_generation(wind_speed, efficiency):
+    print("calculate", wind_speed)
+    sleep(1)
+    power_generation = wind_speed * efficiency
+    return f"Wind Speed: {wind_speed} m/s, Efficiency: {efficiency}, Power Generation: {power_generation:.1f} kW"
+
+
+wind_speed = pn.widgets.FloatSlider(
+    value=5, start=0, end=20, step=1, name="Wind Speed (m/s)"
+)
+
+efficiency = 0.3
+
+calculate_power_fn = pn.bind(
+    calculate_power_generation, wind_speed=wind_speed.param.value_throttled, efficiency=efficiency
+)
+
+pn.Column(
+    wind_speed, calculate_power_fn
+).servable()
+```
+
+Try dragging the slider. Notice that the `calculate_power_generation` function is only run when you release the mouse.
+
+### Binding to bound functions
+
+You may also `bind` to bound functions. This can help you break down you reactivity into smaller, reusable steps.
+
+```{pyodide}
+import panel as pn
+
+pn.extension()
+
+def power_generation(wind_speed, efficiency):
+    print(wind_speed, efficiency)
+    return wind_speed * efficiency
+
+def power_generation_text(wind_speed, efficiency, power):
+    return f"Wind Speed: {wind_speed} m/s, Efficiency: {efficiency}, Power Generation: {power:.1f} kW"
+
+def to_html(text):
+    return f"<strong>{text}<strong>"
+
+wind_speed = pn.widgets.FloatSlider(
+    value=5, start=0, end=20, step=1, name="Wind Speed (m/s)"
+)
+
+efficiency = 0.3
+
+b_power_generation = pn.bind(power_generation, wind_speed=wind_speed, efficiency=efficiency)
+
+b_power_generation_text = pn.bind(power_generation_text, wind_speed, efficiency, b_power_generation)
+
+pn.Column(
+    wind_speed, b_power_generation, b_power_generation_text
+).servable()
+```
+
+```{warning}
+Binding to bound functions can help you to quickly explore your data but its highly inefficient as the results are calculated from scratch for each call.
+```
+
+Try changing the `power_generation` function to:
+
+```python
+def power_generation(wind_speed, efficiency):
+    print(wind_speed, efficiency)
+    return wind_speed * efficiency
+```
+
+Try dragging the `wind_speed` slider. Notice that the `power_generation` function is called twice every time you change the `wind_speed` `value`.
+
+To solve this problem you should add *caching* or use *reactive expressions* (`pn.rx`). You will learn about *reactive expressions* in the next section.
+
+## Triggering Side Effects with `watch`
```

Comment:
Please provide your *best practice* versions of the two apps in this section @philippjfr . Thx.